docs/website/contents/for-developers/NodeTasks.md

```diff
@@ -0,0 +1,305 @@
+
+In brief (see the linked network specification for more details):
+
+ - **ChainSync**:
+   In the context of a caught-up node, the ChainSync mini protocol allows the client to subscribe to updates of the selection of the upstream peer.
+   Such an update might either be a new header[^ChainSync-N2C] (`MsgRollForward`), or a rollback to a point in the past (`MsgRollBackward`).
+
+   The node validates all headers (in particular, this encompasses the validation of the stake-based *election proof*) and maintains a *candidate header chain* for that peer (the result of the messages received so far), which is used to decide which blocks to download.
+
+ - **BlockFetch**:
+   The client can request arbitrary blocks from the server (potentially in batches).
+
+ - **TxSubmission**
+   The server can request transactions from the client.
+
+   Generally, blocks flow from upstream to downstream peers, while transactions flow from downstream to upstream peers (in order to allow nodes without any downstream peers to still submit transactions to the network).
+
+#### Diffusion of headers and blocks
+
+```mermaid
+graph LR
+  subgraph PeerA[Upstream Peer A]
+    PeerAChainSync[ChainSync]
+    PeerABlockFetch[BlockFetch]
+  end
+
+  subgraph PeerB[Upstream Peer B]
+    PeerBChainSync[ChainSync]
+    PeerBBlockFetch[BlockFetch]
+  end
+
+  subgraph PeerC[Downstream Peer C]
+    PeerCChainSync[ChainSync]
+    PeerCBlockFetch[BlockFetch]
+  end
+
+  subgraph PeerD[Downstream Peer D]
+    PeerDChainSync[ChainSync]
+    PeerDBlockFetch[BlockFetch]
+  end
+
+  BlockFetchLogic
+
+  PeerAChainSync -- headers --> BlockFetchLogic
+  BlockFetchLogic -- instructs --> PeerABlockFetch
+  PeerBChainSync -- headers --> BlockFetchLogic
+  BlockFetchLogic -- instructs --> PeerBBlockFetch
+
+  ChainSelectionAndStorageLayer[ChainSelection and storage layer]
+  PeerABlockFetch -- blocks --> ChainSelectionAndStorageLayer
+  PeerBBlockFetch -- blocks --> ChainSelectionAndStorageLayer
+
+  ChainSelectionAndStorageLayer -- headers --> PeerCChainSync
+  ChainSelectionAndStorageLayer -- blocks --> PeerCBlockFetch
+  ChainSelectionAndStorageLayer -- headers --> PeerDChainSync
+  ChainSelectionAndStorageLayer -- blocks --> PeerDBlockFetch
+```
+
+Here, BlockFetchLogic is a centralized component that decides whether and in which order to download certain blocks based on the candidate header chains.
+
+#### Diffusion of transactions
+
+```mermaid
+graph LR
+  subgraph PeerA[Upstream Peer A]
+    PeerATxSubmission[TxSubmission]
+  end
+
+  subgraph PeerB[Upstream Peer B]
+    PeerBTxSubmission[TxSubmission]
+  end
+
+  subgraph PeerC[Downstream Peer C]
+    PeerCTxSubmission[TxSubmission]
+  end
+
+  subgraph PeerD[Downstream Peer D]
+    PeerDTxSubmission[TxSubmission]
+  end
+
+  Mempool
+
+  Mempool --> PeerATxSubmission
+  Mempool --> PeerBTxSubmission
+  PeerCTxSubmission --> Mempool
+  PeerDTxSubmission --> Mempool
+```
+
+## Relaying a new block
```

Comment:
Yeah, do you think this would be useful?
```mermaid
graph LR
  subgraph UpstreamPeers[Upstream peers]
    PeerX[Peer X]
    PeerY[Peer Y]
    PeerZ[Peer Z]
  end

  subgraph DownstreamPeers[Downstream peers]
    PeerA[Peer A]
    PeerB[Peer B]
    PeerC[Peer C]
  end

  TheNode

  PeerX --> TheNode
  PeerY --> TheNode
  PeerZ --> TheNode

  TheNode --> PeerA
  TheNode --> PeerB
  TheNode --> PeerC
```

Reply:
It would be for me, but like I said, it might be obvious for most people. So feel free to skip it.